sycl/source/detail/scheduler/scheduler.hpp

```diff
@@ -745,13 +755,13 @@ class Scheduler {
 
   // Protects stream buffers pool
   std::mutex StreamBuffersPoolMutex;
-  std::map<stream_impl *, StreamBuffers> StreamBuffersPool;
+  std::map<stream_impl *, StreamBuffers *> StreamBuffersPool;
```

Comment:
Would use of `unordered_map` benefit here in general?

Reply:
Agree, there could be potential benefit from unordered_map.